tests/components/gogogate2/test_cover.py

```diff
@@ -0,0 +1,531 @@
+            status=DoorStatus.UNDEFINED,
+            sensor=True,
+            sensorid=None,
+            camera=False,
+            events=0,
+            temperature=None,
+        ),
+        door3=Door(
+            door_id=3,
+            permission=True,
+            name=None,
+            mode=DoorMode.GARAGE,
+            status=DoorStatus.UNDEFINED,
+            sensor=True,
+            sensorid=None,
+            camera=False,
+            events=0,
+            temperature=None,
+        ),
+        outputs=Outputs(output1=True, output2=False, output3=True),
+        network=Network(ip=""),
+        wifi=Wifi(SSID="", linkquality="", signal=""),
+    )
+
+    await component_factory.configure_component()
+    assert hass.states.get("cover.door1") is None
+
+    component_data = await component_factory.run_config_flow(
+        config_data={
+            CONF_IP_ADDRESS: "127.0.0.2",
+            CONF_USERNAME: "user0",
+            CONF_PASSWORD: "password0",
+        }
+    )
+
+    component_data.api.activate.return_value = ActivateResponse(result=True)
+
+    assert hass.states.get("cover.door1").state == STATE_OPEN
+    await hass.services.async_call(
+        COVER_DOMAIN, "close_cover", service_data={"entity_id": "cover.door1"},
+    )
+    await hass.async_block_till_done()
+    component_data.api.close_door.assert_called_with(1)
+    await hass.services.async_call(
+        HA_DOMAIN, "update_entity", service_data={"entity_id": "cover.door1"},
+    )
+    await hass.async_block_till_done()
+    assert hass.states.get("cover.door1").state == STATE_CLOSING
+
+    component_data.data_update_coordinator.api.info.return_value = closed_door_response
+    await component_data.data_update_coordinator.async_refresh()
+    await hass.async_block_till_done()
+    assert hass.states.get("cover.door1").state == STATE_CLOSED
+
+    # Assert mid state changed when new status is received.
+    await hass.services.async_call(
+        COVER_DOMAIN, "open_cover", service_data={"entity_id": "cover.door1"},
+    )
+    await hass.async_block_till_done()
+    component_data.api.open_door.assert_called_with(1)
+    await hass.services.async_call(
+        HA_DOMAIN, "update_entity", service_data={"entity_id": "cover.door1"},
+    )
+    await hass.async_block_till_done()
+    assert hass.states.get("cover.door1").state == STATE_OPENING
+
+    # Assert the mid state does not change when the same status is returned.
+    component_data.data_update_coordinator.api.info.return_value = closed_door_response
+    await component_data.data_update_coordinator.async_refresh()
+    component_data.data_update_coordinator.api.info.return_value = closed_door_response
+    await component_data.data_update_coordinator.async_refresh()
+    await hass.services.async_call(
+        HA_DOMAIN, "update_entity", service_data={"entity_id": "cover.door1"},
+    )
+    await hass.async_block_till_done()
+    assert hass.states.get("cover.door1").state == STATE_OPENING
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5535891)